### PR TITLE
fix(remap): Escape quotes and backslashes when printing VRL strings

### PR DESCRIPTION
**Pull request by pablosichert**
Noticed that as part of https://github.com/timberio/vector/pull/6668.

Otherwise we end up printing values that can not be copy-pasted again into the script or CLI.

### CODE REVIEW

File: lib/vrl/compiler/Cargo.toml [hunk 1]
```diff
@@ -8,10 +8,12 @@ publish = false
 [dependencies]
 diagnostic = { package = "vrl-diagnostic", path = "../diagnostic" }
 parser = { package = "vrl-parser", path = "../parser" }
+shared = { path = "../../shared", default-features = false, features = ["btreemap"] }
```
Comment (jszwedko) [inline lib/vrl/compiler/Cargo.toml:11]: Could this just be a `dev-dependencies` given we only use it for tests?

**Review by jszwedko — APPROVED** [review]
Nice catch! I think this is good to go in as-is, but I am wondering if we should be translating other escape characters like tab, newline, etc.

I see this crate might be an option: https://github.com/euank/snailquote . I suppose it depends on how we support escape sequences when decoding strings.

**Comment by pablosichert** [comment]
> but I am wondering if we should be translating other escape characters like tab, newline, etc.

Good question! I'd say for anything that can be printed and read in as Unicode, can be preserved (e.g. tab). Newline should probably handled, since it breaks the format.

**Review by ktff — APPROVED** [review]
>  Newline should probably handled,

It would be nice to add that here, but any more than that should probably be handled with a crate like the one @jszwedko mentioned which seems like a good next upgrade.

**Comment by jszwedko** [comment]
Agreed, anything that would cause it to fail being parsed should be escaped. That seems like at least "\r" and "\n", but I get the feeling there may be others 😄 I'm unsure about copy pasting other non-printable characters.

Again, this is a strict improvement so I'm fine with it as-is, but it does seem like it'd be worthwhile to take a look at doing something more fancy, maybe that crate, in the future.

**Pull request merged**